server/sonar-server/src/main/java/org/sonar/server/platform/ServerLogging.java

```diff
@@ -64,7 +64,7 @@ public LoggerLevel getRootLoggerLevel() {
    * The directory that contains log files. May not exist.
    */
   public File getLogsDir() {
-    return new File(settings.getString(ProcessProperties.PATH_LOGS));
+    return new File(config.get(ProcessProperties.PATH_LOGS).get());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655)